### PR TITLE
Issue #2681 - Mode choice for specific commute type 4ci

### PR DESCRIPTION
**Pull request by carloscaldas**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2724)

Issue: https://github.com/LBNL-UCB-STI/beam/issues/2681

### CODE REVIEW

File: src/main/java/beam/analysis/StatsFactory.java [hunk 1]
```diff
@@ -2,11 +2,18 @@
 
 import beam.analysis.cartraveltime.PersonAverageTravelTimeAnalysis;
 import beam.analysis.plots.*;
-import beam.analysis.summary.*;
+import beam.analysis.plots.filterevent.ActivitySimFilterEvent;
```
Comment (wrashid) [inline src/main/java/beam/analysis/StatsFactory.java:5]: actitivitySim config parameter missing to enable

Comment (carloscaldas) [reply src/main/java/beam/analysis/StatsFactory.java:5]: @wrashid added. can be checked in the following lines:
https://github.com/LBNL-UCB-STI/beam/pull/2724/files#diff-4d795c860a10b4727cb9f9da894b3eefR244
https://github.com/LBNL-UCB-STI/beam/pull/2724/files#diff-c8b3324eb5c46ac44755ae58b926a3bdR16

**Comment by carloscaldas** [comment]
CI is passing: https://beam-ci.tk/job/beam-4ci/8273/
image

File: src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    val current: Option[PlanElement] = planElements.lift(event.tourIndex - 1)
+    val next: Option[PlanElement] = planElements.lift(event.tourIndex)
+    isHomeOrWorkActivity(current) && isHomeOrWorkActivity(next)
+  }
+
+  private def isHomeOrWorkActivity(event: ReplanningEvent): Boolean = {
+    isHomeOrWorkActivity(event.getEventType)
+  }
+
+  private def isHomeOrWorkActivity(activity: Option[PlanElement]): Boolean = {
```
Comment (REASY) [inline src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala:37]: nitpick: pattern matching can be replaces by `Option[PlanElement].exists()` method

File: src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.matsim.api.core.v01.events.Event
+import org.matsim.api.core.v01.population.{Activity, PlanElement}
+import org.matsim.core.controler.MatsimServices
+
+class ActivitySimFilterEvent(beamConfig: BeamConfig, matsimServices: MatsimServices) extends FilterEvent {
+
+  override def graphNamePreSuffix: String = "_commute"
+
+  override def shouldProcessEvent(event: Event): Boolean = {
+    val isEnabled = beamConfig.beam.exchange.scenario.urbansim.activitySimEnabled
```
Comment (REASY) [inline src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala:16]: nitpick: can be moved to the scope of class fields

File: src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import scala.collection.JavaConverters._
+
+import beam.agentsim.events.{ModeChoiceEvent, ReplanningEvent}
+import beam.sim.config.BeamConfig
+import org.matsim.api.core.v01.events.Event
+import org.matsim.api.core.v01.population.{Activity, PlanElement}
+import org.matsim.core.controler.MatsimServices
+
+class ActivitySimFilterEvent(beamConfig: BeamConfig, matsimServices: MatsimServices) extends FilterEvent {
```
Comment (REASY) [inline src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala:11]: Please, add some tests to verify functionality

File: src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala (hunk 2 above)
Comment (REASY) [reply src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala:37]: As per discussion, you're doing pattern-matching to get the Activity type, `exists` won't work here, so no need to replace it.

File: src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala (hunk 4 above)
Comment (carloscaldas) [reply src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala:12]: added

**Comment by carloscaldas** [comment]
Test!

**Comment by carloscaldas** [comment]
Build succeeded: https://beam-ci.tk/job/beam-4ci/8294/
image

**Pull request merged**